apps/heureka/src/components/common/ServiceImageVersions/index.tsx

```diff
@@ -47,50 +46,32 @@ export const ServiceImageVersions = ({
     })
   const gridColumnCount = displayActions ? COLUMN_COUNT : COLUMN_COUNT - 1
 
-  const selectImageVersion = useCallback(
-    ({ service, imageVersion }: { service: ServiceType; imageVersion: ServiceImageVersion }) => {
-      dispatch({
-        type: ActionType.SelectImageVersion,
-        payload: {
-          service,
-          imageVersion,
-          showPanel: true,
-        },
-      })
-    },
-    [dispatch]
-  )
-
-  const showServiceDetails = useCallback(
-    ({ service, imageVersion }: { service: ServiceType; imageVersion?: ServiceImageVersion }) => {
+  /**
+   * when imageVersions are loaded or the default selection is changed
+   * inform parent to change the highlighted image version in the list
+   */
+  useEffect(() => {
+    if (defaultSelectedImageVersion) {
```

Comment:
(just clarification)
This is looks fgood for me maybe could talk about it more detail, just leave it here for clarification

Reply:
We focussing on Tanstack again we will refactor this components later as talked online